### PR TITLE
refactor: enable -Wconversion warning for edit.c

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/567

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

**Comment by mhinz** [comment]
Our only gcc job complains about a sign conversion issue: https://github.com/neovim/neovim/runs/6226987335?check_suite_focus=true#step%3A15%3A38=

**Comment by dundargoc** [comment]
Damn, should've paid more attention. Thanks for the heads up.

**Comment by dundargoc** [comment]
@mhinz done. Thanks for the heads up.

Also added a cast to string.c which threw an conversion error as well...for some reason. I guess I must've missed it previously.

File: src/nvim/marktree.c [hunk 1]
```diff
@@ -832,10 +832,10 @@ static void itr_swap(MarkTreeIter *itr1, MarkTreeIter *itr2)
   rawkey(itr2).pos = key2.pos;
 }
 
-bool marktree_splice(MarkTree *b, int start_line, int start_col, int old_extent_line,
+bool marktree_splice(MarkTree *b, linenr_T start_line, int start_col, int old_extent_line,
```
Comment (bfredl) [inline src/nvim/marktree.c:835]: hmm, I would rather change this to `int32_t` and add cast at caller

File: src/nvim/marktree.c [hunk 2]
```diff
@@ -583,9 +583,9 @@ void marktree_move(MarkTree *b, MarkTreeIter *itr, int row, int col)
 // itr functions
 
 // TODO(bfredl): static inline?
-bool marktree_itr_get(MarkTree *b, int row, int col, MarkTreeIter *itr)
+bool marktree_itr_get(MarkTree *b, linenr_T row, int col, MarkTreeIter *itr)
```
Comment (bfredl) [inline src/nvim/marktree.c:586]: the same

**Pull request merged**